core/src/main/java/com/datastrato/gravitino/catalog/CatalogManager.java

```diff
@@ -449,10 +452,20 @@ public Catalog alterCatalog(NameIdentifier ident, CatalogChange... changes)
 
                 return newCatalogBuilder.build();
               });
-      return catalogCache.get(
-              updatedCatalog.nameIdentifier(), id -> createCatalogWrapper(updatedCatalog))
-          .catalog;
 
+      catalogCache.invalidate(ident);
+      try {
+        CatalogWrapper updateCatalogWrapper =
+            catalogCache.get(
+                updatedCatalog.nameIdentifier(),
+                id -> updateCatalogWrapper(updatedCatalog, catalogWrapper));
+        return updateCatalogWrapper.catalog;
+      } catch (Exception e) {
```

Comment:
What kind of exception will be thrown here?

Reply:
It depends on methods `createCatalogWrapper` and `buildCatalogWrapper`. Maybe it's 'NoClassFoundError', 'InitializationFailError', or else like `IllegalArgumentException` thrown by `validatePropertyForCreate`.

<img width="1264" alt="image" src="https://github.com/datastrato/gravitino/assets/15794564/db7bcaef-885b-4805-b652-60e79ff21f76">